tests/components/switcher_kis/test_init.py

```diff
@@ -101,56 +106,84 @@ async def test_set_auto_off_service(
         context=Context(user_id=hass_owner_user.id),
     )
 
-    with raises(Unauthorized) as unauthorized_read_only_exc:
+    with raises(UnknownUser) as unknown_user_exc:
         await hass.services.async_call(
             DOMAIN,
             SERVICE_SET_AUTO_OFF_NAME,
             {CONF_ENTITY_ID: SWITCH_ENTITY_ID, CONF_AUTO_OFF: DUMMY_AUTO_OFF_SET},
             blocking=True,
-            context=Context(user_id=hass_read_only_user.id),
+            context=Context(user_id="not_real_user"),
         )
 
-    assert unauthorized_read_only_exc.type is Unauthorized
+    assert unknown_user_exc.type is UnknownUser
 
-    with raises(Unauthorized) as unauthorized_wrong_entity_exc:
-        await hass.services.async_call(
-            DOMAIN,
-            SERVICE_SET_AUTO_OFF_NAME,
-            {
-                CONF_ENTITY_ID: "light.not_related_entity",
-                CONF_AUTO_OFF: DUMMY_AUTO_OFF_SET,
-            },
-            blocking=True,
-            context=Context(user_id=hass_owner_user.id),
-        )
+    service_calls = async_mock_service(
```

Comment:
It's better to patch the library and assert the call to the library mock. The integration will register the service for us so we shouldn't mock that.

Reply:
Thanks, I just copy a another existing test, I fixed both tests to patch the library and assert the call to the library mock.